python/GafferUI/SpreadsheetUI.py

```diff
@@ -1212,6 +1315,276 @@ def __fixedWidth( cls, plugValueWidget ) :
+			for cellPlug in self.__rowsPlug["default"]["cells"] :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
+				if m == sectionName or ( not m and sectionName == "Other" ) :
+					Gaffer.Metadata.registerValue( cellPlug, "spreadsheet:section", newSectionName )
+
+			# Reapply section order for renamed section.
+			self.__assignSectionOrder(
+				self.__rowsPlug,
+				[ newSectionName if n == sectionName else n for n in sectionNames ]
+			)
+
+			# And choose the renamed tab if necessary.
+			if sectionIsCurrent :
+				self._qtWidget().setCurrentIndex( sectionNames.index( sectionName ) )
+
+	def __deleteSection( self, sectionName ) :
+
+		sectionNames = self.sectionNames( self.__rowsPlug )
+
+		with Gaffer.UndoScope( self.__rowsPlug.ancestor( Gaffer.ScriptNode ) ) :
+			# Iterate in reverse to avoid invalidating column indices we're
+			# about to visit.
+			for columnIndex, cellPlug in reversed( list( enumerate( self.__rowsPlug["default"]["cells"] ) ) ) :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
+				if m == sectionName or ( not m and sectionName == "Other" ) :
+					self.__rowsPlug.removeColumn( columnIndex )
+			# Reassign section indices to remove gap.
+			self.__assignSectionOrder(
+				self.__rowsPlug,
+				[ n for n in sectionNames if n != sectionName ]
+			)
+
+	def __moveSection( self, fromSectionName, toSectionName ) :
+
+		sectionNames = self.sectionNames( self.__rowsPlug )
+		sectionIsCurrent = self._qtWidget().tabText( self._qtWidget().currentIndex() ) == fromSectionName
+
+		with Gaffer.UndoScope( self.__rowsPlug.ancestor( Gaffer.ScriptNode ) ) :
+
+			# Move columns
+			for cellPlug in self.__rowsPlug["default"]["cells"] :
+				m = Gaffer.Metadata.value( cellPlug, "spreadsheet:section" )
+				if m == fromSectionName or ( not m and fromSectionName == "Other" ) :
+					Gaffer.Metadata.registerValue( cellPlug, "spreadsheet:section", toSectionName )
+
+			# Reapply section order to remove gaps.
+			newSectionNames = [ n for n in sectionNames if n != fromSectionName ]
+			self.__assignSectionOrder( self.__rowsPlug, newSectionNames )
+
+			if sectionIsCurrent :
+				self._qtWidget().setCurrentIndex( newSectionNames.index( toSectionName ) )
+
+	def __contextMenuRequested( self, pos ) :
+
+		m = IECore.MenuDefinition()
+		sectionName = self._qtWidget().tabText( self._qtWidget().tabAt( pos ) )
+
+		m.append(
+			"/Rename",
+			{
+				"command" : functools.partial( Gaffer.WeakMethod( self.__renameSection ), sectionName )
+			}
+		)
+
+		sectionNames = self.sectionNames( self.__rowsPlug )
+		for toSectionName in sectionNames :
+			if toSectionName == sectionName :
+				continue
+			m.append(
+				"/Move Columns To/{}".format( toSectionName ),
+				{
+					"command" : functools.partial( Gaffer.WeakMethod( self.__moveSection ), sectionName, toSectionName )
+				}
+			)
+
+		m.append( "/__DeleteDivider__", { "divider" : True } )
+
+		m.append(
+			"/Delete",
```

Comment:
I get the logic in deleting the columns from the sheet along with the section, I'm sure people will get used to it (especially as it's undoable). I was just thinking about @gregmassievfx's request:
 
> remove all items from tab (putting them back in the Other tab).

My initial expectation (until I'd renamed 'Other') was that removing a section when I just had `MySection` and `Other` would get me back to where I was before I added a section instead. Ie: no sections and all my columns still.   

If I only have 2 sections, i've not renamed the Other section, the tab bar disappears. If I have renamed it, then I get a single section with whatever name it has.

Reply:
d35774d. As discussed separately, if the only section remaining is "Other" then it auto-hides. There's also a new "Remove Sectioning" menu item to remove all sections and get you back to a big ole list of columns.